eng/common/scripts/job-matrix/README.md

```diff
@@ -59,55 +99,87 @@ jobs:
+
+`MatrixFilters` and `MatrixReplace` allow runtime adjustment of the matrix generation process as can be seen in the source of [GenerateMatrix](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/job-matrix/job-matrix-functions.ps1#L94-L95).  
+See also [Filters](#filters) and [Replace/Modify/Append Values](#replacemodifyappend-values).
 
 ## Matrix config file syntax
 
-Matrix parameters can either be a list of strings, or a set of grouped strings (represented as a hash). The latter parameter
-type is useful for when 2 or more parameters need to be grouped together, but without generating more than one matrix permutation.
+Matrix config file is a [JSON](https://www.json.org/json-en.html) file.
 
-```
+The top-level element in the config is a JSON object having following keys:
+`matrix`, `include`, `exclude` and `displayNames`.
+For explanation of all the top-level keys, see [Matrix JSON config fields](#matrix-json-config-fields) section below.
+Note that `include` and `exclude` have different interpretation than [their
+equivalents in GitHub matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations).
+
+### Matrix syntax
+
+In `matrix`, each key-value pair denotes a parameter and its values.
+
+Each parameter value is either an array of strings, or an object.
+
+If a parameter value is an object, then we say that parameter represents
+a `parameter set group` and the parameter value is a group of all the valid
+parameter sets.
+
+The `parameter set group` kind of parameter is useful for when 2 or more parameters
+need to be grouped together, but without generating more than one matrix combination.
+In such case we group them into one set of given parameter set group.
+
+Grammar of a config file, as an example:
+
+``` yaml
 "matrix": {
   "<parameter1 name>": [ <values...> ],
   "<parameter2 name>": [ <values...> ],
-  "<parameter set>": {
-    "<parameter set 1 name>": {
-        "<parameter set 1 value 1": "value",
-        "<parameter set 1 value 2": "<value>",
+  "<parameter set group>": {
+    "<parameter set group set 1 name>": {
+        "<parameter set 1 key 1": "<value>",
+        "<parameter set 1 key 2": "<value>",
     },
-    "<parameter set 2 name>": {
-        "<parameter set 2 value 1": "value",
-        "<parameter set 2 value 2": "<value>",
+    "<parameter set group set 2 name>": {
+        "<parameter set group set 2 key 1": "<value>",
+        "<parameter set group set 2 key 2": "<value>",
     }
   }
 }
 "include": [ <matrix>, <matrix>, ... ],
 "exclude": [ <matrix>, <matrix>, ... ],
-"displayNames": { <parameter value>: <human readable override> }
+"displayNames": { <parameter value>: <human readable override> },
 ```
 
-See `samples/matrix.json` for a full sample.
+See [`samples/matrix.json`](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/job-matrix/samples/matrix.json) for a full sample.
```

Comment:
This link would need to be updated based on your other PR that moves samples to `eng/common-tests`

Reply:
Yeah, I will update these paths before merging this PR, and after the other PR doing the move is merged.